netsim/tools/nuts/tests/test_bgp.j2

```diff
@@ -0,0 +1,43 @@
+{% set ns = namespace(found=false) %}
+- test_class: TestNapalmBgpNeighbors
+  test_data:
+{% for nn,nd in nodes.items() %}
+{%   for neighbor in nd.bgp.neighbors %}
```

Comment:
Just FYI (no need to change): this works only because the underlying data structure is a Box, creating nd.bgp and nd.bgp.neighbors on demand even if they did not exist before.

Even `nd.bgp is defined` would create nd.bgp, the only safe test is `"bgp" in nd`. However, that doesn't matter as _netlab_ creates a copy of the topology before creating a tool configuration file.

Reply:
I faced an issue that `{% if neighor.ipv4 %}` broke the bgp deployment. I have now tried to make the bgp and neighbor test more stable.